XCUITests/DataManagementTests.swift

```diff
@@ -0,0 +1,43 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import XCTest
+
+class DataManagementTests: BaseTestCase {
+
+    //Testing the entries are correctly added and deleted. This a termporary behaviour. This test will be changed after releasing Firefox 15.x and Bug 1499074 will be fixed.
+    func testWebSiteDataEnterFirstTime() {
+        navigator.goto(WebsiteDataSettings)
+        XCTAssertEqual(app.tables.cells.count, 2)
+        XCTAssertTrue(app.tables.staticTexts["Clear All Website Data"].exists)
+        navigator.performAction(Action.AcceptClearAllWebsiteData)
+        XCTAssertEqual(app.tables.cells.count, 1)
+        navigator.createNewTab()
+        navigator.goto(WebsiteDataSettings)
+        XCTAssertEqual(app.tables.cells.count, 2)
+    }
+    
+    // Testing the search bar, search results and 'Show More' option.
+    func testWebSiteDataOptions() {
+        navigator.openURL("google.com")
+        navigator.openURL("youtube.com")
+        navigator.openURL("twitter.com")
+        navigator.openURL("facebook.com")
+        navigator.goto(WebsiteDataSettings)
+        navigator.performAction(Action.TapOnFilterWebsites)
+        app.typeText("google")
+        waitforExistence(app.tables["Search results"])
+        XCTAssertEqual(app.tables["Search results"].cells.count, 2)
+        navigator.performAction(Action.TapOnFilterWebsites)
+        app.searchFields["Filter Sites"].tap()
```

Comment:
I added that line because I didn't know how to clear the search field. So after navigator.performAction(Action.TapOnFilterWebsites) in the search bar there  is "google" typed. After app.searchFields["Filter Sites"].tap() the "google" text is selected and when doing app.typeText("foo") the "google' text is cleared and the new search term is typed "foo". 

If app.searchFields["Filter Sites"].tap() is removed it will just type "googlefoo" which I guess would also be good as we are testing for a non-existing search term.

Reply:
Oh I see...but yes, it will work and the test will be clearer without that line...